doc/rst/language/spec/tuples.rst

```diff
@@ -714,6 +715,286 @@ where a comma-separated list of components is valid.
+      var i: int;
+      var r = new R(0);
+
+      // The tuple variable `tup` copies `a`, `i`, and `r`.
+      var tup = (a, i, r);
+
+      a[0] = 1;
+      i = 2;
+      r.x = 3;
+
+      // This will output (0, 0, (x = 0)).
+      writeln(tup);
+
+   .. BLOCK-test-chapeloutput
+
+      (0, 0, (x = 0))
+
+The tuple variable ``tup`` will make a copy of the array ``a``, the record
+``r``, and the module integer ``i``.
+
+When these three variables are subsequently modified, changes to them are
+not reflected in ``tup`` when it is written to standard output.
+
+.. _Tuple_Argument_Behavior:
+
+Tuple Argument Behavior
+-----------------------
+
+A tuple argument to a routine may be either a light tuple or a heavy tuple
+depending on its argument intent.
+
+If the tuple argument has the default argument intent, then it is a light
+tuple and some of its elements may be captured by `ref` depending on
+their default argument intent.
+
+If the tuple argument has the `in` intent, then it is a heavy tuple and all
+of its elements are captured by value, as though each element has the `in`
+intent.
+
+If a routine argument is a tuple with the default argument intent and a
+heavy tuple (such as a tuple variable) is passed to it, the heavy tuple
+will be silently converted into a light tuple.
+
+A conversion from light to heavy tuple also occurs when a light tuple
+(such as a tuple expression) is passed to a tuple argument with the `in`
+intent.
+
+   *Example (tuple-argument-behavior.chpl)*.
+
+Consider the following example:
+
+   .. code-block:: chapel
+
+      record R { var x: int; }
+
+      var modTup = (0, new R(0));
+
+      proc lightTupleArg(light) {
+
+        // Modify the module variable `modTup`.
+        modTup = (3, new R(6));
+
+        // Should print (0, (x = 6)).
+        writeln(light);
+
+        heavyTupleArg(light);
+
+        // Should still print (0, (x = 6)).
+        writeln(light);
+      }
+
+      proc heavyTupleArg(in heavy) {
+        heavy = (64, new R(128));
+      }
+
+      lightTupleArg(modTup);
+
+   .. BLOCK-test-chapeloutput
+
+      (0, (x = 6))
+      (0, (x = 6))
+
+The argument ``tup`` of the routine ``lightTupleArg`` is a light tuple due
+to the default argument intent. When the module variable ``modTup`` is
+passed to ``lightTupleArg``, its first element is copied while its second
+element is passed as though it were `const ref`.
+
+When the tuple argument ``tup`` is passed to ``heavyTupleArg``, a copy of it
+is made because the argument ``heavy`` is a heavy tuple due to the `in`
+intent.
+
+   *Example (tuple-argument-ref-intent.chpl)*.
+
+Tuple arguments with the `ref` intent are considered to be a special form
+of light tuple where every element is passed by reference.
+
+If a tuple argument has the `ref` intent, then actual arguments are
+restricted to heavy tuples (a tuple variable or a returned tuple). Each
+individual element behaves as though it was an argument with the `ref`
+intent.
```

Comment:
I'd remove these two paragraphs and replace them with this:

Tuple arguments with `ref` intent are references to heavy tuples.
Actual arguments are restricted to heavy tuples (a tuple variable or a returned tuple).
Since the argument itself is passed by `ref`, the entire tuple will refer to a tuple from the call site.